app/controllers/admin/assets_controller.rb

```diff
@@ -15,17 +15,17 @@ def unspam
 
     def spam
       AssetCommand.new(@asset).spam_and_soft_delete_with_relations
-      redirect_to admin_assets_path(filter_by: :is_spam)
+      redirect_to admin_assets_path(filter_by: :is_spam) unless params[:no_redirect]
```

Comment:
This `no_redirect` pattern here suggests to me that we're working around a problem and there's probably a clearer solution. Can you describe what the need is here? Are these controller methods not called via ajax in the new view?

Reply:
It was a really bad user experience when i'm on an `user#show` page and was being redirected to `assets#spam` if I wanted to spam some of user's assets.
I'm working on a user, I don't want to be redirected anywhere.

Reply:
Gotcha! It seems like the spamming should be happening via ajax?...

Reply:
Well... we do some stimulusjs actions but then redirect to `assets/index(spammed)`[ in the controller](https://github.com/sudara/alonetone/blob/master/app/controllers/admin/assets_controller.rb#L16-L19)
```
    def spam
      AssetCommand.new(@asset).spam_and_soft_delete_with_relations
      redirect_to admin_assets_path(filter_by: :is_spam)
    end
```
Which made sense when we did it. I forsee us needed more bulk actions with minimum redirects, but for now, leave as is?